src/utils.js

```diff
@@ -10,30 +10,53 @@ export function getNetworkName(chainId) {
 
 export function rpcResult(response) {
   // Some providers don’t wrap the response
-  if (response && response.jsonrpc) {
+  if (response && 'jsonrpc' in response) {
     if (response.error) {
       throw new Error(response.error)
     }
-    return response.result
+    return response.result || ''
   }
-  return response
+  return response || ''
+}
+
+async function sendCompat(ethereum, method, params) {
+  // Old MetaMask versions are using sendAsync(), and web3.js 1.x is
+  // overwriting send() with sendAsync(), despite them being incompatible.
+  // There is no way to detect that send() follows the correct implementation,
+  // so we use sendAsync() when it is available.
+  if (ethereum.sendAsync && ethereum.selectedAddress) {
+    return new Promise((resolve, reject) => {
+      ethereum.sendAsync(
+        { method, params, from: ethereum.selectedAddress },
+        (err, result) => {
+          if (err) {
+            reject(err)
+          } else {
+            resolve(result)
+          }
+        }
+      )
+    }).then(rpcResult)
+  }
+
+  return ethereum.send(method, params).then(rpcResult)
```

Comment:
Wondering, would it be better to prefer `ethereum.send()` over `sendAsync()`?

In the case of web3.js overwriting `send()`, it shouldn't even be available, right?

Reply:
No `send()` is available with the web3.js overwrite, but it gets replaced by `sendAsync()` (which is what the `send()` signature used to be until recently).

We could try to detect what `send()` it is, but I don’t think there is any non-hacky, reliable way to do that.

Reply:
:( that makes me really sad.

I wish we named the new `send()` something different then...

Reply:
> Can we clarify with a comment about this?

Done in https://github.com/aragon/use-wallet/pull/21/commits/990c7ebdb3d045a49c012579fe4d9ba47dc227b3